src/cpp/r/RUtil.cpp

```diff
@@ -72,6 +80,79 @@ bool versionTest(const std::string& comparator, const std::string& version)
 
 } // anonymous namespace
 
+void setenv(const std::string& key, const std::string& value)
+{
+   core::system::setenv(key, value);
+
+#ifdef _WIN32
+   // NOTE: Required on Windows as R links to a static copy
+   // of libc and libgcc, and so has a separate environment
+   // block from the rsession.exe executable itself.
+   Error error = r::exec::RFunction("base:::Sys.setenv")
+         .addParam(key, value)
+         .call();
+
+   if (error)
+      LOG_ERROR(error);
+#endif
+}
+
+std::string getenv(const std::string& key)
+{
+   std::string value;
+   Error error = r::exec::RFunction("base:::Sys.getenv")
+         .addParam(key)
+         .call(&value);
+
+   if (error)
+      LOG_ERROR(error);
+
+   return value;
+}
+
+namespace {
+
+void modifySystemPath(const std::string& pathEntry, bool prepend)
+{
+   std::string oldPath = getenv("PATH");
+   std::string newPath = prepend
+         ? fmt::format("{}{}{}", pathEntry, kPathSeparator, oldPath)
+         : fmt::format("{}{}{}", oldPath, kPathSeparator, pathEntry);
+
+#ifdef _WIN32
+   std::replace(newPath.begin(), newPath.end(), '/', '\\');
+#endif
+
+   boost::regex reDuplicateSeparators(kPathSeparator "+");
+   boost::regex_replace(newPath, reDuplicateSeparators, kPathSeparator);
```

Comment:
FYI I still see in some instances duplicate separators, so I'm not positive this bit is working

Reply:
> I think you need to assign it to a result?

You are right; I incorrectly thought this routine took its first argument by reference and so mutated the argument in place. Thank you for catching that!

> also isn't a + needed in between kPathSeparator and "+" ?

In this case no; a feature of C / C++ is that sequential string tokens are automatically concatenated; e.g. `"a" "b" "c"` becomes `"abc"`. Since `kPathSeparator` is a macro expanding to a string we can take advantage of the same feature here.